[NET-10985] Fix bug where imagePullSecrets were not set up for Gateways (#4316)
hashicorp/consul-k8s: .changelog/4316.txt

```release-note:bug
api-gateway: `global.imagePullSecrets` are now configured on the `ServiceAccount` for `Gateways`.

> Note: the referenced image pull Secret(s) must be present in any namespace that the `Gateway` is deployed to.

Reviewer: ```suggestion
Note: the referenced image pull Secret(s) must be present in the same namespace the `Gateway` is deployed to.
```
